triangle/example.java

```diff
@@ -0,0 +1,53 @@
+public class Triangle {
+
+    private double side1;
+    private double side2;
+    private double side3;
+
+    public Triangle(double side1, double side2, double side3) throws TriangleException {
+        this.side1 = side1;
+        this.side2 = side2;
+        this.side3 = side3;
+
+        if (allSidesAreZero() || hasImpossibleSides() || violatesTriangleInequality()) {
+            throw new TriangleException();
```

Comment:
From a more Java-y standpoint (and ignoring the checked vs unchecked exception debate), I would probably just want to use an `IlllegalArgumentException`.

Also, it might be helpful if the exception included a message indicating which condition was violated (and possibly which specific side in question is in violation).